urlb3290c8e9e_craigonic_jasl_tgz-pJ8-DriverJ8/src/jasl/ui/data/Player.java

```diff
@@ -0,0 +1,307 @@
+
+	public String toString()
+	{
+		return _name;
+	}
+
+	/**
+	 * Return the nationality of the majority of the units that a player
+	 * controls.
+	 *
+	 * @return a <CODE>String</CODE> specifying the nationality.
+	 */
+
+	public String nationality()
+	{
+		return _nationality.toString();
+	}
+
+	/**
+	 * Return the turn that the units initially managed by a player will
+	 * enter the game.
+	 *
+	 * @return an <CODE>int</CODE> specifying the entry turn.
+	 */
+
+	public int entryTurn()
+	{
+		return _entryTurn;
+	}
+
+	/**
+	 * Return one of the units managed by a player.
+	 *
+	 * @param index the (zero-based) index of the <A HREF="../../counters/Unit.html">Unit</A> to be accessed
+	 *
+	 * @return the <CODE>Unit</CODE> at the specified index.
+	 *
+	 * @throws IndexOutOfBoundsException in the case where index is less than
+	 * zero or greater than the number of units in the list.
+	 *
+	 * @see #unitList
+	 */
+
+	public String unit(int index)
+	{
+		String unitToBeReturned;
+
+		try
+		{
+			unitToBeReturned = _unitList.get(index);
+		}
+
+		catch (IndexOutOfBoundsException exception)
+		{
+			throw new IndexOutOfBoundsException(Messages.buildErrorMessage(CLASS_NAME,
+			                                                               "unit",
+			                                                               Messages.INVALID_PARAMETER_MSG + index));
+		}
+
+		return unitToBeReturned;
+	}
+
+	/**
+	 * Return a list of descriptions of the units managed by a player.
+	 *
+	 * @return a <CODE>List</CODE> of String items specifying the labels associated with
+	 * the player's units.
+	 */
+
+	public List<String> unitList()
+	{
+		return Collections.unmodifiableList(_unitList);
+	}
+
+	// Public update methods
+
+	/**
+	 * Place a new unit under a player's command.
+	 *
+	 * @param unit the <A HREF="../../counters/Unit.html">Unit</A> to be added
+	 *
+	 * @throws NullPointerException in the case of a null object.
+	 */
+
+	public void addUnit(String unit)
+	{
+		if (null == unit)
+		{
+			throw new NullPointerException(Messages.buildErrorMessage(CLASS_NAME,
+			                                                          "addUnit",
+			                                                          Messages.NULL_PARAMETER_MSG));
+		}
+
+		_unitList.add(unit);
+	}
+
+	/**
+	 * Remove a unit under a player's command.
+	 *
+	 * If the unit to be removed is currently managing (i.e. portaging)
+	 * other Units (e.g. support weapons), those items will be placed back
+	 * into the general list after the specified unit is deleted.
+	 *
+	 * @param index the (zero-based) index of the <A HREF="../../counters/Unit.html">Unit</A> to be removed
+	 *
+	 * @throws IndexOutOfBoundsException in the case where index is less than
+	 * zero or greater than the number of units in the list.
+	 */
+
+	public void deleteUnit(int index)
+	{
+		String unitToBeDeleted;
+
+		try
+		{
+			unitToBeDeleted = _unitList.get(index);
+		}
+
+		catch (IndexOutOfBoundsException exception)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)